src/AlgebraicGeometry/Curves/ProjectivePlaneCurve.jl

```diff
@@ -0,0 +1,323 @@
+  S = standard_covering(C)
+  i = findfirst(i->!iszero(P[i]), 1:3)
+  Ca = S[i]
+  Q = Ca(P)
+  L = tangent_lines(Ca, Q)
+  D = Dict{ProjectivePlaneCurve, Int}()
+  if !isempty(L)
+    D = Dict(ProjectivePlaneCurve(help_homogene_line(R, ambient_coordinate_ring(Ca), defining_equation(x), i)) => L[x] for x in keys(L))
+  end
+  return D
+end
+
+@doc raw"""
+     intersection_multiplicity(C::S, D::S, P::AbsProjectiveRationalPoint) where S <: ProjectivePlaneCurve
+
+Return the intersection multiplicity of `C` and `D` at `P`.
+"""
+function intersection_multiplicity(C::S, D::S, P::AbsProjectiveRationalPoint) where S <: ProjectivePlaneCurve
+  P in ambient_space(C) || error("The point needs to be in a projective two dimensional space")
+  SC = standard_covering(C)
+  SD = standard_covering(D)
+  i = findfirst(x->!iszero(P[x]), 1:3)
+  return intersection_multiplicity(SC[i], SD[i], SC[i](dehomogenization(P,i)))
+end
+
+################################################################################
+
+
+@doc raw"""
+     are_transverse(C::S, D::S, P::AbsProjectiveRationalPoint) where S <: ProjectivePlaneCurve
+
+Return `true` if `C` and `D` intersect transversally at `P` and `false` otherwise.
+"""
+function are_transverse(C::S, D::S, P::AbsProjectiveRationalPoint) where S <: ProjectivePlaneCurve
+  P in C && P in D || return false
+  any(P in i for i in common_components(C,D)) && return false
+  is_smooth_at(C, P) && is_smooth_at(D, P) && intersection_multiplicity(C, D, P) == 1
+end
+
+################################################################################
+
+@doc raw"""
+    arithmetic_genus(C::ProjectivePlaneCurve)
+
+Return the arithmetic genus of `C`.
+
+# Examples
+```jldoctest
+julia> T, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"]);
+
+julia> C = plane_curve(y^2 * z - x^3 - x * z^2)
+Projective plane curve
+  defined by 0 = x^3 + x*z^2 - y^2*z
+
+julia> arithmetic_genus(C)
+1
+```
+"""
+function arithmetic_genus(C::ProjectivePlaneCurve)
+  H = hilbert_polynomial(C)
+  return -ZZ(coeff(H, 0)) + 1
+end
+
+################################################################################
+
+@doc raw"""
+    geometric_genus(C::ProjectivePlaneCurve)
+
+Return the geometric genus of `C`.
+
+# Examples
+```jldoctest
+julia> R, (x,y,z) = graded_polynomial_ring(QQ, ["x", "y", "z"]);
+
+julia> C = plane_curve(z*x^2-y^3)
+Projective plane curve
+  defined by 0 = x^2*z - y^3
+
+julia> geometric_genus(C)
+0
+
+```
+"""
+function geometric_genus(C::ProjectivePlaneCurve{<:QQField})
+  I = vanishing_ideal(C)
+  singular_assure(I)
+  return ZZ(Singular.LibNormal.genus(I.gens.S)::Int)
+end
+
+function geometric_genus(C::ProjectivePlaneCurve)
+  # buggy
+  A = homogeneous_coordinate_ring(C)
+  L = normalization(A)
+  m = length(L)
+  pa = zero(ZZ)
+  for i in 1:m
+    J = L[i][1].I
+    T, _ = grade(parent(J[1]))
+    V = T.(gens(J))
+    JJ = ideal(T, V)
+    B = quo(T, JJ)
+    H = hilbert_polynomial(B[1])
+    pa = pa - ZZ(coeff(H, 0))
```

Comment:
Do we have something like `constant_coefficient` as a getter here to make this more readable?